### PR TITLE
hazelcast-cpp-client/all: bump deps

### PR DESCRIPTION
**Pull request by ericLemanissier**
Specify library name and version:  **hazelcast-cpp-client/***

Generated and committed by [Conan Center Bump Deps](https://github.com/ericLemanissier/conan-center-index-bump-deps)
Find more updatable recipes in the [GitHub Pages](https://ericlemanissier.github.io/conan-center-index-bump-deps/)

---

- [ ] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [ ] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [ ] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7246036132)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7246036132)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7264819958/attempts/1)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7264819958/attempts/2)

File: recipes/hazelcast-cpp-client/all/conanfile.py [hunk 1]
```diff
@@ -44,7 +43,7 @@ def layout(self):
         cmake_layout(self, src_folder="src")
 
     def requirements(self):
-        self.requires("boost/1.79.0", transitive_headers=True, transitive_libs=True)
+        self.requires("boost/1.81.0", transitive_headers=True, transitive_libs=True)
```
Comment (uilianries) [inline recipes/hazelcast-cpp-client/all/conanfile.py:46]:
```suggestion
        self.requires("boost/1.83.0", transitive_headers=True, transitive_libs=True)
```

Comment (ericLemanissier) [reply recipes/hazelcast-cpp-client/all/conanfile.py:46]: It used to fail with boost 1.83.0 https://c3i.jfrog.io/c3i/misc/logs/pr/21754/3-linux-gcc/hazelcast-cpp-client/5.2.0//5d9d2780a74d4113b0223cf73d9be450956502e3-test.txt
But test program simplification may work around this error.

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7328926689/attempts/1)

**Comment by ericLemanissier** [comment]
The build is now finished. You can take a look at the results [here](https://github.com/ericLemanissier/proof-of-conan/actions/runs/7328926689/attempts/2)

**Review by uilianries — APPROVED** [review]

**Review by toge — APPROVED** [review]

**Pull request merged**